### ISSUE TITLE
many assert (os.path.exists(....)) test errors on python3 branch

### ISSUE THREAD

**Issue by az0** [bleachbit/bleachbit#759]
The system and revision information here is the same as https://github.com/bleachbit/bleachbit/issues/757

When I run `make tests`, I get 21 failures, 39 errors, and 26 skipped. Below is the tail of the error log. Many errors look the same with `assert (os.path.exists(extended_path(filename)))`

```
======================================================================
FAIL: test_whitelisted_posix_symlink (tests.TestFileUtilities.FileUtilitiesTestCase)
Symlink test for whitelisted_posix()
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/bleachbit/tests/common.py", line 160, in wrapper
    return f(*args, **kwargs)
  File "/tmp/bleachbit/tests/TestFileUtilities.py", line 792, in test_whitelisted_posix_symlink
    realpath = self.write_file('real')
  File "/tmp/bleachbit/tests/common.py", line 116, in write_file
    assert (os.path.exists(extended_path(filename)))
AssertionError

======================================================================
FAIL: test_wipe_contents (tests.TestFileUtilities.FileUtilitiesTestCase)
Unit test for wipe_delete()
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/bleachbit/tests/TestFileUtilities.py", line 844, in test_wipe_contents
    filename = self.write_file('bleachbit-test-wipe', b'abcdefghij' * 12345)
  File "/tmp/bleachbit/tests/common.py", line 116, in write_file
    assert (os.path.exists(extended_path(filename)))
AssertionError

======================================================================
FAIL: test_wipe_name (tests.TestFileUtilities.FileUtilitiesTestCase)
Unit test for wipe_name()
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/bleachbit/tests/TestFileUtilities.py", line 880, in test_wipe_name
    filename = self.write_file('bleachbit-test-wipe' + '0' * 50)
  File "/tmp/bleachbit/tests/common.py", line 116, in write_file
    assert (os.path.exists(extended_path(filename)))
AssertionError

======================================================================
FAIL: test_fakelocalizationdirs (tests.TestUnix.UnixTestCase)
Create a faked localization hierarchy and clean it afterwards
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/bleachbit/tests/common.py", line 160, in wrapper
    return f(*args, **kwargs)
  File "/tmp/bleachbit/tests/TestUnix.py", line 185, in test_fakelocalizationdirs
    self.write_file(path)
  File "/tmp/bleachbit/tests/common.py", line 116, in write_file
    assert (os.path.exists(extended_path(filename)))
AssertionError

======================================================================
FAIL: test_is_running (tests.TestUnix.UnixTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/bleachbit/tests/common.py", line 160, in wrapper
    return f(*args, **kwargs)
  File "/tmp/bleachbit/tests/TestUnix.py", line 105, in test_is_running
    self.assertTrue(is_running(exe))
AssertionError: False is not true

======================================================================
FAIL: test_many_patterns (tests.TestWinapp.WinappTestCase)
Test a cleaner like Steam Installers and related performance improvement
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/bleachbit/tests/TestWinapp.py", line 439, in test_many_patterns
    common.touch_file(tmp_fn)
  File "/tmp/bleachbit/tests/common.py", line 182, in touch_file
    assert(os.path.exists(filename))
AssertionError

----------------------------------------------------------------------
Ran 167 tests in 74.622s

FAILED (failures=21, errors=39, skipped=26)
make: *** [Makefile:107: tests] Error 1
```

**Comment by mkhon** [comment]
python3 branch is not compatible with python2. You need to run tests using python3. Please let me know which environment do you have.

**Comment by az0** [comment]
This is Python 3.7.3 on Ubuntu 19.04. 

Originally I ran `make tests`, and it defaulted to python3 because $COVERAGE on Makefile line 108 defaults to python3 where COVERAGE is defined on line 22.

I also tried `python3 m unittest discover -p Test*.py -v`, and the results are the same

```
======================================================================
FAIL: test_many_patterns (tests.TestWinapp.WinappTestCase)
Test a cleaner like Steam Installers and related performance improvement
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/bleachbit/tests/TestWinapp.py", line 439, in test_many_patterns
    common.touch_file(tmp_fn)
  File "/tmp/bleachbit/tests/common.py", line 182, in touch_file
    assert(os.path.exists(filename))
AssertionError

----------------------------------------------------------------------
Ran 167 tests in 58.155s

FAILED (failures=21, errors=39, skipped=26)
```

**Comment by mkhon** [comment]
I've tested everything on Ubuntu 18.04.

19.04 is not LTS - I will take a look a bit later as I don't have 19.04 at hand right now.

**Comment by az0** [comment]
> Ubuntu 19.04 is not LTS - I will take a look a bit later as I don't have 19.04 and/or Python 3.7 at hand right now.

How about I merge merge this PR,  and then I try to get Travis to test multiple Python versions, something like 3.6 and 3.7?

**Comment by az0** [comment]
I modified `.travis.yml` to test Python 3.5, 3.6, and 3.7. 

Versions 3.5 and 3.6 finish OK, but 3.7 has an error: [Travis  Job #1741.3](https://travis-ci.org/bleachbit/bleachbit/jobs/642052374).

```
Ran 180 tests in 63.163s
FAILED (failures=21, errors=37, skipped=28)
```

**Comment by mkhon** [comment]
Fixed in #763

**Comment by az0** [comment]
Yes, fixed. Thanks
